packages/mongodb-chatbot-server/src/processors/makeVerifiedAnswerGenerateUserPrompt.ts

```diff
@@ -10,6 +11,11 @@ export interface MakeVerifiedAnswerGenerateUserPromptParams {
    */
   findVerifiedAnswer: FindVerifiedAnswerFunc;
 
+  /**
+    Format or modify the verified answer before displaying it to the user.
+   */
+  onVerifiedAnswerFound?: (verifiedAnswer: VerifiedAnswer) => VerifiedAnswer;
```

Comment:
what was the reason for this addition?

Reply:
I don't love it. It comes down to:

1. We don't enforce reference metadata on the verified answer objects so we want to add them on the server
2. Verified Answers on the server use a different code path than regular answers, so we need to handle the reference metadata in two place
3. This is the way I chose to handle it for the verified answer code path